pkg/service/connections.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package service
+
+import (
+	"errors"
+	"net"
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/pkg/datapath/sockets"
+	lb "github.com/cilium/cilium/pkg/loadbalancer"
+)
+
+var opSupported = true
+
+func (s *Service) destroyConnectionsToBackend(be *lb.Backend) {
+	if !opSupported {
+		return
+	}
+	log.Infof("debug-aditi destroying connections to backend %v", be.L3n4Addr)
+	var (
+		family   uint8
+		protocol uint8
+	)
+	ip := net.IP(be.L3n4Addr.AddrCluster.Addr().AsSlice())
+	l4Addr := be.L3n4Addr.L4Addr
+
+	switch be.L3n4Addr.Protocol {
+	case lb.UDP:
+		protocol = unix.IPPROTO_UDP
+	default:
+		return
+	}
+	if be.L3n4Addr.IsIPv6() {
+		family = syscall.AF_INET6
+	} else {
+		family = syscall.AF_INET
+	}
+	// Filter pod connections load-balanced to the passed service backend.
+	//
+	// When pod traffic is load-balanced to service backends, the cilium datapath
+	// records entries in the sock rev nat map that store the pod socket cookie
+	// (unique identifier in the kernel) along with the destination backend ip/port.
+	checkSockInRevNat := func(id netlink.SocketID) bool {
+		log.Infof("debug-aditi check rev nat %v %d", id, id.Cookie)
+		cookie := uint64(id.Cookie[1])
+		cookie = cookie<<32 + uint64(id.Cookie[0])
+
+		return s.lbmap.ExistsSockRevNat(cookie, id.Destination, id.DestinationPort)
+	}
+
+	err := s.backendConnectionHandler.Destroy(sockets.SocketFilter{
+		Family:    family,
+		Protocol:  protocol,
+		DestIp:    ip,
+		DestPort:  l4Addr.Port,
+		DestroyCB: checkSockInRevNat,
+	})
+	if errors.Is(err, unix.EOPNOTSUPP) {
```

Comment:
We might want to log something for errors other than `unix.EOPNOTSUPP`. Otherwise, we won't know what happened here.

Reply:
Sure!